### ISSUE TITLE
Windows CI test failures

### ISSUE THREAD

**Issue by patzim** [rakudo/rakudo#3071]
Latest [AppVeyor build](https://ci.appveyor.com/project/rakudo/rakudo/branch/master/job/omr5d37bskj4ujug) has this set of test failures. There are some new failures.

> Test Summary Report
> -------------------
> t\02-rakudo\14-revisions.t                                    (Wstat: 0 Tests: 2 Failed: 1)
>   Failed test:  2
> t\04-nativecall\06-struct.t                                   (Wstat: 0 Tests: 29 Failed: 0)
>   Parse errors: Bad plan.  You planned 34 tests but ran 29.
> t\05-messages\02-errors.t                                     (Wstat: 0 Tests: 37 Failed: 8)
>   Failed tests:  1, 5, 13-14, 16-17, 25-26
> t\05-messages\03-errors.t                                     (Wstat: 0 Tests: 25 Failed: 6)
>   Failed tests:  6, 10, 15-17, 25
> t\05-messages\moar\01-errors.t                                (Wstat: 0 Tests: 1 Failed: 1)
>   Failed test:  1
> t\09-moar\General_Category__extracted-DerivedGeneralCategory.t (Wstat: 0 Tests: 1 Failed: 0)
>   TODO passed:   1
> t\09-moar\Line_Break__LineBreak.t                             (Wstat: 0 Tests: 2 Failed: 0)
>   TODO passed:   1
> Files=99, Tests=1755, 167 wallclock secs ( 0.58 usr +  0.14 sys =  0.72 CPU)
> Result: FAIL
> NMAKE : fatal error U1077: 'C:\strawberry\perl\bin\perl.exe' : return code '0x1'
> Stop.
> Command exited with code 2

**Comment by AlexDaniel** [comment]
The PR for fixing struct issue slipped through the cracks, and I only merged it now: https://github.com/MoarVM/MoarVM/pull/1138

What's up with everything else I'm not sure. @vrurg, anything looks familiar?

**Comment by vrurg** [comment]
I don't currently see a point for a new ticket, but I have problems with perl6-m.exe in my Win7 VM. It looks to me like the binary forks and executes in background. It complicates things a lot, especially when run under Far manager. 

I'm also confident that his is what causes the test failures. Every failed test I checked is expecting exit code 1. But because perl6-m.exe goes into background, shell exits with exit code 0 and the actual code of perl6 process is getting lost.

**Comment by patzim** [comment]
I fell victim to [this](https://stackoverflow.com/questions/7004687/os-exec-on-windows). There is no real `exec()` on Windows. I'll try to do a CreateProcess with inheriting all handles and waiting on it.

**Comment by patzim** [comment]
These should be fixed with #3078 and [NQP#566](https://github.com/perl6/nqp/pull/566).

**Comment by vrurg** [comment]
BTW, tested on my Win7 VM, confirm passing of `make test`.
